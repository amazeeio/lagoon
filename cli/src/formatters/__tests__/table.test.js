// @flow

import formatAsTable from '../table';

describe('table formatter', () => {
  it('should format data with 0 data rows correctly', () => {
    const formatted = formatAsTable([['Name', 'Customer']]);
    expect(formatted).toMatchSnapshot();
  });

  it('should format data with 1 data row correctly', () => {
    const formatted = formatAsTable([
      ['Name', 'Customer'],
      ['test-project-table-1', 'test-customer-table-1'],
    ]);
    expect(formatted).toMatchSnapshot();
  });

  it('should format data with > 1 data rows correctly', () => {
    const formatted = formatAsTable([
      ['Name', 'Customer'],
      ['test-project-table-1', 'test-customer-table-1'],
      ['test-project-table-2', 'test-customer-table-2'],
    ]);
    expect(formatted).toMatchSnapshot();
  });
});
