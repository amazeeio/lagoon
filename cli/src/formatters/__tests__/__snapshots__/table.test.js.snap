// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table formatter should format data with > 1 data rows correctly 1`] = `
"[38;2;132;132;132mName[39m                  [38;2;132;132;132mCustomer[39m
test-project-table-1  test-customer-table-1
test-project-table-2  test-customer-table-2"
`;

exports[`table formatter should format data with 0 data rows correctly 1`] = `"[38;2;132;132;132mName[39m  [38;2;132;132;132mCustomer[39m"`;

exports[`table formatter should format data with 1 data row correctly 1`] = `
"[38;2;132;132;132mName[39m      test-project-table-1
[38;2;132;132;132mCustomer[39m  test-customer-table-1"
`;
